src/types.d.ts

```diff
@@ -110,4 +106,99 @@ export interface Blockstore {
    * Close the store
    */
   close: () => Promise<Void>
-}
\ No newline at end of file
+}
+
+export interface Config {
+  Addresses: AddressConfig
+  API?: APIConfig,
+  Profiles?: string
+  Bootstrap: string[]
+  Discovery: DiscoveryConfig
+  Datastore?: DatastoreConfig
+  Identity?: IdentityConfig
+  Keychain?: KeychainConfig
+  Pubsub?: PubsubConfig
+  Swarm?: SwarmConfig
+  Routing?: RoutingConfig
+}
+
+/**
+ * Contains information about various listener addresses to be used by this node
+ */
+export interface AddressConfig {
+  API?: string
+  RPC?: string
+  Delegates?: string[]
+  Gateway?: string
+  Swarm: string[]
+}
+
+export interface APIConfig {
+  HTTPHeaders?: Record<string, string>
+}
+
+export interface DiscoveryConfig {
+  MDNS: MDNSDiscovery
```

Comment:
is the discovery config required per module?

Reply:
The options are different between MDNS and WebRTCStar so it seems so, yes